pkg/util/log/log.go

```diff
@@ -488,32 +513,51 @@ func logFormatWithError(logLevel seelog.LogLevel, bufferFunc func(), logFunc fun
 	return err
 }
 
-func logContext(logLevel seelog.LogLevel, bufferFunc func(), logFunc func(string), message string, depth int, context ...interface{}) {
+func logContext(logLevel seelog.LogLevel, bufferFunc func(), logFunc func(string, seelog.LoggerInterface), message string, depth int, context ...interface{}) {
 	if Logger != nil && Logger.inner != nil && Logger.shouldLog(logLevel) {
 		msg := Logger.scrub(message)
 		Logger.l.Lock()
 		defer Logger.l.Unlock()
 		Logger.inner.SetContext(context)
 		Logger.inner.SetAdditionalStackDepth(defaultStackDepth + depth) //nolint:errcheck
-		logFunc(msg)
+		logFunc(msg, Logger.inner)
+		Logger.inner.SetContext(nil)
+		Logger.inner.SetAdditionalStackDepth(defaultStackDepth) //nolint:errcheck
+	} else if Logger != nil && Logger.verbose != nil && !Logger.shouldLog(logLevel) {
```

Comment:
I think it could be cleaner to have this in an "else" of the `Logger.shouldLog` condition more directly

Reply:
It's more clear to separate checking `Logger.verbose` and `Logger.inner`